db/migrate/20170329064703_create_user_emails.rb

```diff
@@ -0,0 +1,31 @@
+class CreateUserEmails < ActiveRecord::Migration
+  def up
+    create_table :user_emails do |t|
+      t.integer :user_id
+      t.string :email, limit: 513
+      t.timestamps
+    end
+    add_index :user_emails, :user_id
+    execute "CREATE UNIQUE INDEX index_user_emails_on_email ON user_emails ((lower(email)));"
+    execute "INSERT INTO user_emails (user_id, email, created_at, updated_at) SELECT id, email, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP FROM users WHERE users.id != -1"
+    change_table :users do |t|
+      t.remove :email
```

Comment:
You'll have to delay dropping this column otherwise everything blows up during deployment on setups where migrations are run while web servers quite still kept running. You can have a look at https://github.com/discourse/discourse/blob/c53eeaaa6806b2e872f6ee21d4db01d5c437c152/lib/column_dropper.rb

Also, I'll prefer if we keep this column around first so that we don't lose everyone's email if something does terribly wrong during the migration. The column can be dropped maybe a month or two after things have stabilized.